src/modules/dashboard/sagas/utils/safeHelpers.ts

```diff
@@ -280,6 +281,92 @@ export const getTransferFundsData = async (
   ]);
 };
 
+const getParsedJSONOrArrayFromString = (
+  parameter: string,
+): (string | number)[] | null => {
+  try {
+    const arrayResult = JSON.parse(parameter);
+    return arrayResult.map((value) => {
+      if (Number.isInteger(value)) {
+        return new BigNumber(value).toString();
+      }
+      return value;
+    });
+  } catch (err) {
+    return null;
+  }
+};
+
+const isArrayParameter = (parameter: string): boolean =>
+  /(\[\d*])+$/.test(parameter);
+
+const extractMethodArgs = (
+  signature: string,
+  transaction: Record<string, any>,
+) => ({ name = '' }) => {
+  if (isArrayParameter(transaction[name || signature])) {
```

Comment:
Do we store the signature as a key in the SafeTransaction object? I've not come across this, I've only noticed that we store the function name as the key and the input data as the value.

Reply:
True, I think this is a leftover from when I was testing some things

Reply:
Gonna have to update this anyways with the merge of #3952